### PR TITLE
Introducing the msRuleCleaner configuration.

### PR DESCRIPTION
**Pull request by todor-ivanov**
This is the initial PR for creating the MSRuleCleaner configuration.

### CODE REVIEW

File: exporters/manage [hunk 1]
```diff
@@ -167,6 +167,8 @@ reqmgr2ms_monitors()
         </dev/null 2>&1 | rotatelogs $LOGDIR/ms-monitor_exporter-%Y%m%d-`hostname -s`.log 86400 >/dev/null 2>&1 &
     process_monitor.sh /data/srv/state/reqmgr2ms/ms-output.pid ms_output ":18245" 15 \
         </dev/null 2>&1 | rotatelogs $LOGDIR/ms-output_exporter-%Y%m%d-`hostname -s`.log 86400 >/dev/null 2>&1 &
+    process_monitor.sh /data/srv/state/reqmgr2ms/ms-rulecleaner.pid ms_rulecleaner ":18244" 15 \
```
Comment (amaltaro) [inline exporters/manage:170]: @todor-ivanov this port will have to be opened on the backend running MSRuleCleaner. And I guess Valentin will have to do something else on the Prometheus side. Please discuss those with Imran and Valentin as well.

File: reqmgr2ms/config-ruleCleaner.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+LOG_REPORTER = "reqmgr2ms_ruleCleaner"
+ROOTDIR = __file__.rsplit('/', 3)[0]
+AMQ_HOST_PORT = [('cms-mb.cern.ch', 61313)]
+
+# load AMQ credentials
+sys.path.append(path.join(ROOTDIR, 'auth/reqmgr2ms'))
+from ReqMgr2MSSecrets import USER_AMQ, PASS_AMQ, AMQ_TOPIC
+
+RUCIO_ACCT = "wma_prod"
+RULE_LIFETIME = 30 * 24 * 60 * 60  # 30 days
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:26]: RULE_LIFETIME, SEND_NOTIFICATION and ENABLE_DATA_PLACEMENT are not needed in this service.

File: reqmgr2ms/config-ruleCleaner.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+data = views.section_('data')
+data.object = 'WMCore.MicroService.Service.RestApiHub.RestApiHub'
+# The couch host is defined during deployment time.
+data.couch_host = COUCH_URL
+data.couch_wmstats_db = "wmstats"
+data.manager = 'WMCore.MicroService.Unified.MSManager.MSManager'
+data.reqmgr2Url = "%s/reqmgr2" % BASE_URL
+data.limitRequestsPerCycle = 500
+# Allows or not some alert notifications to be sent to the production-admin egroup
+data.sendNotification = SEND_NOTIFICATION
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:79]: not needed

File: reqmgr2ms/config-ruleCleaner.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+data.verbose = True
+data.interval = 60 * 60 * 1  # run it every hour
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:81]: I'd set the interval to every 8h

File: reqmgr2ms/config-ruleCleaner.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+data.services = ['ruleCleaner']
+data.defaultGroup = "DataOps"
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:83]: not needed

File: reqmgr2ms/config-ruleCleaner.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+data.excludeDataTier = []
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:84]: not needed

**Review by amaltaro — CHANGES_REQUESTED** [review]
Todor, I've pointed a few configuration parameters that are irrelevant for this microservice. Can you please remove those?

In addition to that, prometheus monitoring needs to be set with Imran/Valentin.

File: exporters/manage (hunk 1 above)
Comment (todor-ivanov) [reply exporters/manage:170]: done

File: reqmgr2ms/config-ruleCleaner.py (hunk 2 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py:26]: done

File: reqmgr2ms/config-ruleCleaner.py (hunk 3 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py:79]: Done

File: reqmgr2ms/config-ruleCleaner.py (hunk 4 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py:81]: Done

File: reqmgr2ms/config-ruleCleaner.py (hunk 6 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py:84]: Done

File: reqmgr2ms/config-ruleCleaner.py (hunk 5 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py:83]: Done

File: reqmgr2ms/config-ruleCleaner.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+data.object = 'WMCore.MicroService.Service.RestApiHub.RestApiHub'
+# The couch host is defined during deployment time.
+data.couch_host = COUCH_URL
+data.couch_wmstats_db = "wmstats"
+data.manager = 'WMCore.MicroService.Unified.MSManager.MSManager'
+data.reqmgr2Url = "%s/reqmgr2" % BASE_URL
+data.limitRequestsPerCycle = 500
+# Allows or not some alert notifications to be sent to the production-admin egroup
+data.verbose = True
+data.interval = 60 * 60 * 8  # run it every hour
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:75]: bad comment, but I didn't say anything :)

File: reqmgr2ms/config-ruleCleaner.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+# REST interface
+data = views.section_('data')
+data.object = 'WMCore.MicroService.Service.RestApiHub.RestApiHub'
+# The couch host is defined during deployment time.
+data.couch_host = COUCH_URL
+data.couch_wmstats_db = "wmstats"
+data.manager = 'WMCore.MicroService.Unified.MSManager.MSManager'
+data.reqmgr2Url = "%s/reqmgr2" % BASE_URL
+data.limitRequestsPerCycle = 500
+# Allows or not some alert notifications to be sent to the production-admin egroup
```
Comment (amaltaro) [inline reqmgr2ms/config-ruleCleaner.py:73]: another unneeded comment

File: reqmgr2ms/config-ruleCleaner.py (hunk 7 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py]: good you spotted that 
Done

File: reqmgr2ms/config-ruleCleaner.py (hunk 8 above)
Comment (todor-ivanov) [reply reqmgr2ms/config-ruleCleaner.py]: Thanks for spotting this leftover too 
Done

**Review by amaltaro — APPROVED** [review]
Thanks Todor, it's looking good now!

**Comment by vkuznet** [comment]
@todor-ivanov , the changes look fine to me, and, yes we will need to open port 18244 on a node to allow prometheus to scrape the metrics. Once it is done, please open corresponding [CMSMONIT](https://its.cern.ch/jira/projects/CMSMONIT) Jira ticket and provide name of the node and port to use.  In case of k8s you'll need to create a new k8s yaml file for this service and then we'll plug it in. For that see example how it is done for other reqmgr microservices, e.g. [ms-monitoring](https://github.com/dmwm/CMSKubernetes/blob/master/kubernetes/cmsweb/services/reqmgr2ms-monitor.yaml) and make new PR against CMSKubernetes. The new k8s yaml file should include your service name, ports, etc.

File: exporters/manage (hunk 1 above)
Comment (muhammadimranfarooqi) [reply exporters/manage:170]: @todor-ivanov @amaltaro 
I have opened 18244 port on backends.

Comment (todor-ivanov) [reply exporters/manage:170]: Thank you @muhammadimranfarooqi

**Comment by amaltaro** [comment]
@todor-ivanov two other actions are required from your side:
1. create the CMSKubernetes yaml file, as pointed out by Valentin above (against this repo: https://github.com/dmwm/CMSKubernetes)
2. create the `preprod` and `prod` respective microservice configuration (against gitlab repo in: https://gitlab.cern.ch/cmsweb-k8s/services_config/-/tree/preprod)

Please link those to this PR once you get that done.
Imran, from my side, we can proceed with the next testbed deployment before Todor implement those.

**Pull request merged**